### PR TITLE
removes .git from build layer, emulates GH build

### PR DESCRIPTION
**Pull request by umputun**
_this related to the discussion in #269_

Remove .git from the docker build context and pass both GITHUB_REF and GITHUB_SHA from the make. It uses an [updated baseimage](https://github.com/umputun/baseimage/commit/40677b819d9b4b2082a5d0e826a46a312a88dfc8) defaulting to "local" if no git repo

### CODE REVIEW

**Comment by umputun** [comment]
@nagimov @bnoctis pls take a look

**Comment by paskal** [comment]
Doesn't it break the scenario where I just download the git repository and run `docker build`, producing `local` image instead of `master`? Even with `make rundev` in this branch I get `remark42 local-20220601T17:52:14`, I wonder if that's expected.

**Comment by umputun** [comment]
yes, it does. it produces a "local" version unless it runs with `make docker`

**Comment by paskal** [comment]
After last changes, `make rundev` still produces `local` build for me.

**Comment by umputun** [comment]
> After last changes, `make rundev` still produces `local` build for me.

probably your local compose missing this part i have added https://github.com/umputun/remark42/pull/1375/commits/eafff01c5555dce96fbc8cf412930c9988820deb#diff-e1a5cf1531a00e0c8773ac834afc54983202fa6f7268566193ac6d09f45495d6R18

**Comment by bnoctis** [comment]
I'll let @nagimov have the say, LGTM though

**Comment by nagimov** [comment]
TLDR: LGTM
<hr>

Unrelated build issues:

- when building via `docker build -t remark42 --build-arg SKIP_FRONTEND_TEST=true --build-arg SKIP_BACKEND_TEST=true .` or `make docker`, getting the following error due to unset `$BUILDPLATFORM`. It seems like `Dockerfile` always expects `$BUILDPLATFORM` to be set, which only comes from `dockerx` make-target (via `--platform`), and not from `docker` make-target.
```
...
Step 15/52 : FROM --platform=$BUILDPLATFORM node:16.13.2-alpine as build-frontend-deps
failed to parse platform : "" is an invalid component of "": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument
make: *** [Makefile:14: docker] Error 1
```

- fixed with `sed -i 's/--platform=$BUILDPLATFORM//' Dockerfile`
- builds ok `docker build -t remark42 --build-arg SKIP_FRONTEND_TEST=true --build-arg SKIP_BACKEND_TEST=true .`, local instance `http://127.0.0.1/web/` works as expected, properly tagged with `version=gitless-docker-eafff01-YYYYMMDDTHH:MM:SS`
- `docker build -t remark42 .` gives a few failed backend tests, all frontend tests pass

**Review by paskal — APPROVED** [review]
